src/base/camera.h

```diff
@@ -7,24 +7,41 @@
 
 #include "matrix.h"
 #include "ray.h"
+#include "scene_object.h"
 
 namespace gm {
 
 /// A camera class that implements a camera with perspective projection.
 /// When generating primary camera rays the convention is to form rays
 /// at the image plane and then transform them using the cameraToWorld
 /// matrix provided in this class.
-class PerspectiveCamera {
+class PerspectiveCamera : public SceneObject {
  public:
   PerspectiveCamera(){};
 
   PerspectiveCamera(const Vector3f &position, const Vector3f &lookAt,
                     const Vector3f &up, size_t imageWidth, size_t imageHeight,
                     float fov);
 
+  // This constructor is based on the information that is included in a glTF
+  // file.
+  PerspectiveCamera(const Vector3f &location, const Quaternionf &rotation,
+                    const float &fov, const std::string &name);
+
   // Compute a new camera ray for the given raster space coordinate
   Ray generate_ray(uint32_t xPos, uint32_t yPos);
 
+  // Allow setting the image dimensions
+  void setImageSize(const size_t &width, const size_t &height) {
```

Comment:
Where does this get called? Couldn't find it in the code but I assume you took care of it somehow

Reply:
It is not called right now. However the image size is not stored in the gltf-file. Thus it can't be set when the camera object is constructed. I assumed it would be a command-line parameter later.

Reply:
We agreed to keep this in.